tool/src/webapp/js/forum-summary.js

```diff
@@ -0,0 +1,59 @@
+var itemsToShow;
+$(function() {
+	//escape markup
+	String.prototype.escapeHTML = function(){
+		return (this.replace(/&/g, '&amp;').replace(/>/g, '&gt;').replace(/</g, '&lt;').replace(/"/g, '&quot;'));
+	};
+	itemsToShow = $('.forum-summary-div').find($('#numberOfConversations')).val();
+	//set the 'numberofconversations' in the hidden field for edit screen
+	$('#numberOfMessages').val(itemsToShow);
+	showForums();
+});
+
+function showForums(){
+	var forumsUrl = $('.forum-summary-site-url').text().replace(/'/g,"");
+	forumsUrl += '.json?n=' + itemsToShow; 
+	var messagesArray = [];
+	var errorText;
+	//Get forums
+	$.ajaxSetup({ cache: false });	//Disable caching for ajax requests
+	$.getJSON(forumsUrl , function(data_forum_collection) {
+		$(data_forum_collection["forums_collection"]).each(function(fc_index, fc_value){
+			var author;
+			if(fc_value['modifiedBy'].indexOf('(') > -1){
+				author = fc_value['modifiedBy'].split('(')[0];
+			}else{
+				author = fc_value['modifiedBy'];
+			}
+			//added forumId , topicId and messageId into the messageArray to link the message into the forum tool
+			var messageArray = [fc_value["forumId"],fc_value['entityTitle'].escapeHTML(),fc_value['entityURL'],fc_value['lastModified'], author,fc_value['topicId'],fc_value['messageId']];
+			messagesArray.push(messageArray);
+		});
+		outputForums(messagesArray);
+		setTimeout('showForums()', 300000);
+	});
+}
+function outputForums(messagesArray){
+	var toolHref = $('.forum-summary-view-url').text().replace(/'/g,"");
+	var text_for_forums = '<div class="forumSummaryHeaderDiv"><h3 class="forumSummaryHeader"><a href="'+toolHref+'" style="text-decoration: none; color: #444" target="_top" title ="Latest forum posts">'+msg("simplepage.forum-header-title")+'</a></h3></div>';
+	if(messagesArray.length == 0){
+		text_for_forums += '<p>'+msg("simplepage.forum-summary-no-message")+'</p>';
+	}
+	else{
+		messagesArray.sort(function(a, b) {
+			return b[3]-a[3]; //inverse sort by [3]=lastModified
```

Comment:
Ah sorry I see the code in `showForums()` building the array, but why not use an object rather than an array as it makes the code much easier to read.